dask/typing.py

```diff
@@ -91,30 +93,36 @@ def __call__(
         raise NotImplementedError("Inheriting class must implement this method.")
 
 
-@runtime_checkable
-class DaskCollection(Protocol):
-    """Protocol defining the interface of a Dask collection."""
-
-    @abc.abstractmethod
-    def __dask_graph__(self) -> Graph:
-        """The Dask task graph.
+def all_collections_newstyle(collections: Iterable[Any]) -> bool:
+    import dask
+    from dask.delayed import Delayed
+
+    filtered_collections = [
+        a
+        for a in collections
+        if dask.is_dask_collection(a)
+        # Delayed always returns True when checked if instance of a Protocol
+        and not isinstance(a, Delayed)
+    ]
+    if filtered_collections and all(
+        isinstance(a, NewDaskCollection) for a in filtered_collections
+    ):
+        return True
+    elif any(isinstance(a, NewDaskCollection) for a in filtered_collections):
+        raise RuntimeError(
+            "Provided multiple collections to dask.compute but mixed old and new style collections."
+        )
+    return False
 
-        The core Dask collections (Array, DataFrame, Bag, and Delayed)
-        use a :py:class:`~dask.highlevelgraph.HighLevelGraph` to
-        represent the collection task graph. It is also possible to
-        represent the task graph as a low level graph using a Python
-        dictionary.
 
-        Returns
-        -------
-        Mapping
-            The Dask task graph. If the instance returns a
-            :py:class:`dask.highlevelgraph.HighLevelGraph` then the
-            :py:func:`__dask_layers__` method must be implemented, as
-            defined by the :py:class:`~dask.typing.HLGDaskCollection`
-            protocol.
+@runtime_checkable
+class DaskGraph(Protocol):
```

Comment:
Graphs can (and likely will) look more complicated than this protocol. However, this is the minimal protocol that is required to submit things to distributed using pickle